pkg/server/server.go

```diff
@@ -42,8 +42,11 @@ func NewServer(host string, port int, insecureSkipTLS bool, backends ...string)
 		return nil, err
 	}
 	// comment/uncomment to switch between debug and release mode
-	//gin.SetMode(gin.ReleaseMode)
+	gin.SetMode(gin.ReleaseMode)
 	r := gin.New()
+	// by default gin will trust all request headers that contain alternative client IP
+	// https://pkg.go.dev/github.com/gin-gonic/gin#Engine.SetTrustedProxies
+	r.SetTrustedProxies(nil)
```

Comment:
We need to set the `TrustedPlatform` if we want to get the real client IP address. Otherwise by setting the trusted proxies to`nil`, the gin `context.ClientIP()` will likely return the IP of the AWS load balancer that sits in front of the OpenShift route.

```suggestion
	r.SetTrustedProxies(nil)
	r.TrustedPlatform = "X-Forwarded-For"
```

`X-Forwarded-For` is the standard forwarding header that AWS adds to their load balancers.

Reply:
Updated.
For the record, the backend servers were still getting the original header including the "X-Forwarded-For" and were logging the IP from it. The difference is now the sprayproxy also logs the IP from the header instead of the original one. Thanks.